File: 0661-image-smoother/0661-image-smoother.java
class Solution {
    public int[][] imageSmoother(int[][] img) {
        int rows = img.length;
        int cols = img[0].length;

        int[][] res = new int[rows][cols];
        int[] directions = {-1, 0, 1};

        for (int r = 0; r < rows; r++) {
            for (int c = 0; c < cols; c++) {
                int total = 0;
                int count = 0;

                for (int dr : directions) {
                    for (int dc : directions) {
                        int nr = dr + r;
                        int nc = dc + c;
                        if (nr >= 0 && nr < rows && nc >= 0 && nc < cols) {
                            total += img[nr][nc];
                            count++;
                        } 
                    }
                }
                res[r][c] = total / count;
            }
        }
        return res;
    }
}

File: 0661-image-smoother/0661-image-smoother.py
class Solution:
    def imageSmoother(self, img: List[List[int]]) -> List[List[int]]:
        
        rows = len(img)
        cols = len(img[0])

        res = [[0] * cols for _ in range(rows)]
        for r in range(rows):
            for c in range(cols):
                total = 0
                count = 0
                for dr in (-1, 0, 1):
                    for dc in (-1, 0, 1):
                        nr, nc = dr + r, c + dc
                        if 0 <= nr < rows and 0 <= nc < cols:
                            count += 1
                            total += img[nr][nc]
                res[r][c] = total // count
        
        return res
                


                

